icu4c/source/i18n/double-conversion-utils.h

```diff
@@ -54,16 +56,23 @@
 #define DOUBLE_CONVERSION_NO_RETURN __attribute__((noreturn))
 #endif
 #endif
-#ifndef UNREACHABLE
+#ifndef DOUBLE_CONVERSION_UNREACHABLE
 #ifdef _MSC_VER
 void DOUBLE_CONVERSION_NO_RETURN abort_noreturn();
 inline void abort_noreturn() { abort(); }
-#define UNREACHABLE()   (abort_noreturn())
+#define DOUBLE_CONVERSION_UNREACHABLE()   (abort_noreturn())
```

Comment:
Not necessary right now, but I wonder if we might want to consider patching this to change it to use the ICU macro `UPRV_UNREACHABLE`, incase somebody customized that macro in ICU to do something other than call `abort()`.

It looks like the two macros are mostly the same, with the only difference being that the double-conversion macro is annotated as no-return for MSVC.
I think that this was done in the double-conversion code to support older versions of Visual Studio (ex: 2010), but I think ICU requires VS2015 due to the C++11 support.

Reply:
Good thought.  I liked changing the ASSERT macro because it eliminated the inclusion of an extra header file.  I don't think changing this one would do the same.  Let's just leave it the way it is; less likely to break things, as you said about the return value semantics.